pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java

```diff
@@ -437,6 +440,28 @@ void testMessageExpiryWithTimestampNonRecoverableException() throws Exception {
 
     }
 
+    @Test
+    public void testIncorrectClientClock() throws Exception {
```

Comment:
It looks like the test can't cover the issue?

<img width="1686" alt="image" src="https://github.com/apache/pulsar/assets/12592133/ca5869e7-4663-460f-9b3a-4e076d84a588">

Reply:
It looks like the issue is we can't use `assertTrue(monitor.expireMessages(maxTTLSeconds));`
It doesn't mean there are messages is expired. It will return true if there is no in-progress expiration task.

Maybe we can try to check the backlog of the 

```
assertEquals(c1.getNumberOfEntriesInBacklog(true), 0);
```